Hw7 jdi

andrei-mikhailovskii opened the pull request.

andrei-mikhailovskii committed:
add PropertiesExctractor.java
add User.java entity
add JdiTest.java test

andrei-mikhailovskii committed:
add JdiSite.java
add TestSetup.java
add HomePage.java
add MetalColorsPage.java
add test.properties

andrei-mikhailovskii committed:
add hw7 profile to pom.xml
changed login method in JdiSite.java
add getters to User.java
refactored HomePage.java

andrei-mikhailovskii committed:
add MetalColorsForm.java form
refactored MatalColorsPage, JdiTest

andrei-mikhailovskii committed:
add Jackson dependency to pom.xml
add the following files: DataProviderClass, JsonDataRoot, JsonDataSet, JsonParserFile, MetalColorEntity, Main(temporary class)
add JDI_ex8_metalsColorsDataSet.json to resources

andrei-mikhailovskii committed:
fixed error with data provider in test class

andrei-mikhailovskii committed:
wrote fill method and assertion for Metals & Colors page
added Enums for checkboxes on Metals & Colors page
added ResultSection component for Metals & Colors page
removed temporary JsonDataRoot and Main files
made code compliant with Check Style plugin

shoggy commented on pom.xml:
```diff
@@ -21,7 +21,7 @@
             <groupId>org.seleniumhq.selenium</groupId>
             <artifactId>selenium-java</artifactId>
-            <version>3.14.0</version>
+            <version>3.141.59</version>
```
please move version to properties section
\+ same action for other dependencies

andrei-mikhailovskii replied:
moved all versions to properties section

shoggy commented on src/test/java/com/epam/tc/hw7/JsonParserFile.java:
```diff
@@ -0,0 +1,32 @@
+@UtilityClass
+public class JsonParserFile {
+
+    public LinkedHashMap fileToMap() {
```
```suggestion
    public LinkedHashMap<KeyType, ValueType> fileToMap() {
```

andrei-mikhailovskii replied:
changed public LinkedHashMap fileToMap() to LinkedHashMap<String, MetalColorEntity> fileToMap()

shoggy commented on src/test/java/com/epam/tc/hw7/JsonParserFile.java:
```diff
@@ -0,0 +1,32 @@
+                    });
+            return  map;
+
+        } catch (IOException exception) {
+            System.out.println("JSON parsing error: " + exception);
+        }
```
It would be better to throw error here
Because we want to know that configuration was not successful and test is broken, instead of `passes`

andrei-mikhailovskii replied:
removed try-with-resources construction, added throw exception and closing of FileReader

shoggy commented on src/test/java/com/epam/tc/hw7/DataProviderClass.java:
```diff
@@ -0,0 +1,26 @@
+
+public class DataProviderClass {
+
+    LinkedHashMap<String, MetalColorEntity> map = JsonParserFile.fileToMap();
```
Good candidate to become local variable

andrei-mikhailovskii replied:
made it local variable

shoggy commented on src/test/java/com/epam/tc/hw7/entities/UserEntity.java:
```diff
@@ -0,0 +1,26 @@
+@Setter
+public class UserEntity {
+
+    public static final UserEntity ROMAN_IOVLEV = new UserEntity(
+            PropertiesExtractor.getUsername(),
+            PropertiesExtractor.getPassword(),
+            PropertiesExtractor.getCorrectUsername());
```
Please move constant User to other class with constants
DTO should stay only DTO and should not store it's instances.

andrei-mikhailovskii replied:
moved constant User to HomePage

shoggy commented on src/test/java/com/epam/tc/hw7/entities/UserEntity.java:
```diff
@@ -0,0 +1,26 @@
+            PropertiesExtractor.getPassword(),
+            PropertiesExtractor.getCorrectUsername());
+
+    private String userName;
+    private String password;
+    private String correctFullUserName;
```
Candidates to become `final`

andrei-mikhailovskii replied:
made them final

andrei-mikhailovskii committed:
moved versions to properties section in pom.xml
changed try-catch to throw exception in JsonParserFile and DataProviderClass respectively

andrei-mikhailovskii committed:
moved constant user ROMAN_IOVLEV from UserEntity to HomePage
made userName, password, correctFullUserName final in UserEntity

shoggy commented on src/test/java/com/epam/tc/hw7/JsonParserFile.java:
```diff
@@ -0,0 +1,29 @@
+
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        FileReader reader = new FileReader("src/test/resources/JDI_ex8_metalsColorsDataSet.json");
+
+        LinkedHashMap<String, MetalColorEntity> map = objectMapper
+                .readValue(reader, new TypeReference<LinkedHashMap<String, MetalColorEntity>>() {
+                    });
+
+        reader.close();
```
Here it is accepted, but in general please use try-with-resources for closables

andrei-mikhailovskii committed:
Merge branch 'main' into hw7-jdi
